python/astro_metadata_translator/translators/sdss.py

```diff
@@ -0,0 +1,205 @@
+        return value
+
+    @cache_translation
+    def to_datetime_end(self):
+        # Docstring will be inherited. Property defined in properties.py
+        return self.to_datetime_begin() + self.to_exposure_time()
+
+    @cache_translation
+    def to_location(self):
+        """Calculate the observatory location.
+
+        Returns
+        -------
+        location : `astropy.coordinates.EarthLocation`
+            An object representing the location of the telescope.
+        """
+
+        # Look up the value since files do not have location
+        value = EarthLocation.of_site("apo")
+
+        return value
+
+    @cache_translation
+    def to_observation_type(self):
+        """Calculate the observation type.
+
+        Returns
+        -------
+        typ : `str`
+            Observation type. Normalized to standard set.
+        """
+        obstype_key = "FLAVOR"
+        if not self.is_key_ok(obstype_key):
+            return "none"
+        obstype = self._header[obstype_key].strip().lower()
+        self._used_these_cards(obstype_key)
+        return obstype
+
+    @cache_translation
+    def to_tracking_radec(self):
+        # Docstring will be inherited. Property defined in properties.py
+        radecsys = ("RADECSYS",)
+        radecpairs = (("RA", "DEC"),)
+        return tracking_from_degree_headers(self, radecsys, radecpairs, unit=u.deg)
+
+    @cache_translation
+    def to_altaz_begin(self):
+        # Docstring will be inherited. Property defined in properties.py
+        az = self._header['AZ']
+        alt = self._header['ALT']
+        # It appears SDSS defines azimuth as increasing from South through East
+        # This translates to North through East
+        az = (-az + 180.)%360.
+        altaz = AltAz(az * u.deg, alt * u.deg,
+                      obstime=self.to_datetime_begin(), location=self.to_location())
+        self._used_these_cards('AZ', 'ALT')
+        return altaz
+
+    @cache_translation
+    def to_boresight_airmass(self):
+        # Docstring will be inherited. Property defined in properties.py
+        try:
+            return altaz_from_degree_headers(self, (("ALT", "AZ"),),
+                                             self.to_datetime_begin()).secz.value  # This is an estimate
+        except Exception as e:
+            if self.to_observation_type() != "science":
+                return None  # allow airmass to be unset in non-science observations
+            raise(e)
+
+    @cache_translation
+    def to_detector_exposure_id(self):
+        # Docstring will be inherited. Property defined in properties.py
+        exposure_id = self.to_exposure_id()
+        if exposure_id is None:
+            return None
+        return int("{:07d}{:02d}".format(exposure_id, self.to_detector_num()))
+
+    @cache_translation
+    def to_detector_group(self):
+        # Docstring will be inherited. Property defined in properties.py
+        if self.is_keyword_defined(self._header, 'CAMCOL'):
```

Comment:
I think you want `if self.is_key_ok("CAMCOL")` here. The `is_keyword_defined` class method is only used for `can_translate` since that's also a class method.